### PR TITLE
backup: support object lock in s3

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/11292
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: br/backup-storage-S3.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ summary: Learn how to use BR to back up data to and restore data from Amazon S3
 
 The Backup & Restore (BR) tool supports using Amazon S3 or other Amazon S3-compatible file storages as the external storage for backing up and restoring data.
 
+> **Note:**
+>
+> To back up data to an S3 bucket with object locking enabled, you need to use TiDB v6.3.0 or later.
```
Comment (3pointer) [inline br/backup-storage-S3.md:12]: `object lock` seems to be proprietary. https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html

Comment (shichun-0415) [inline br/backup-storage-S3.md:12]:
```suggestion
> To back up data to an S3 bucket with Object Lock enabled, you need to use TiDB v6.3.0 or later.
```

Comment (shichun-0415) [reply br/backup-storage-S3.md]: Fixed in 1bd7552

File: br/backup-storage-S3.md [hunk 2]
```diff
@@ -7,6 +7,10 @@ summary: Learn how to use BR to back up data to and restore data from Amazon S3
 
 The Backup & Restore (BR) tool supports using Amazon S3 or other Amazon S3-compatible file storages as the external storage for backing up and restoring data.
 
+> **Note:**
+>
+> To back up data to an S3 bucket with Object Lock enabled, you need to use TiDB v6.3.0 or later.
```
Comment (TomShawn) [inline br/backup-storage-S3.md:12]:
```suggestion
> To back up data to an S3 bucket with Object Lock enabled, make sure that your TiDB cluster is v6.3.0 or later.
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by 3pointer — APPROVED** [review]

**Pull request merged**